packages/contracts/tasks/helpers/benchmarks.ts

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable no-console */
+import { Keypair, Message } from "maci-domainobjs";
+
+import { Poll } from "../../typechain-types";
+
+/**
+ * Utility function to send a batch of messages to a poll contract
+ * @dev This function will attempt to send a batch of messages to a poll contract
+ * and if it fails, it will attempt to send a smaller batch size
+ * @param pollContract - The poll contract to send the messages to
+ * @param message - The message to send
+ * @param keypair - The keypair to sign the message with
+ * @param batchSize - The number of messages to send
+ */
+export async function publishBatch(
+  pollContract: Poll,
+  message: Message,
+  keypair: Keypair,
+  batchSize: number,
+): Promise<void> {
+  const messageBatch = Array.from({ length: batchSize }, () => message.asContractParam());
+  const pubKeyBatch = Array.from({ length: batchSize }, () => keypair.pubKey.asContractParam());
+
+  try {
+    const tx = await pollContract.publishMessageBatch(messageBatch, pubKeyBatch);
+    const receipt = await tx.wait();
+    console.log(`Gas used: ${receipt?.gasUsed.toString()} wei\n`);
+    console.log(`Tx: ${tx.hash}\n`);
+  } catch (err) {
+    console.log(`Failed to submit a batch of ${batchSize} messages\n`);
+    if (batchSize > 1) {
+      const newBatchSize = batchSize - 1;
+      console.log(`Trying with ${newBatchSize} messages\n`);
+      await publishBatch(pollContract, message, keypair, newBatchSize);
+    } else {
+      console.error("Unable to submit even a single message\n");
+    }
+  }
```

Comment:
It's better to avoid using catch for logic handling. Let's use static call to check the right batch size. Ideally, it can be done using binary search.

Reply:
good call, will do that :)

Reply:
seems like with static call it always gives Result(0) [] and I can't seem to find a way to determine whether it'd succeed or not. How have you used static call before?

```ts
    const res = await pollContract.publishMessageBatch.staticCallResult(testMessageBatch, testPubKeyBatch);
```

Reply:
It should work like this
```
await pollContract.publishMessageBatch.staticCall(...args).catch(() => undefined);
```